query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -754,6 +801,30 @@ abstract class QuantileOverTimeChunkedFunction(funcParams: Seq[FuncArgs],
 
 }
 
+abstract class MedianAbsoluteDeviationOverTimeChunkedFunction(var medianAbsoluteDeviationResult: Double = Double.NaN,
+                                                               var values: Buffer[Double] = Buffer.empty[Double])
+  extends ChunkedRangeFunction[TransientRow] {
+  override final def reset(): Unit = { medianAbsoluteDeviationResult = Double.NaN; values = Buffer.empty[Double] }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    val counter = values.length
+    val (weight, upperIndex, lowerIndex) = QuantileOverTimeFunction.calculateRank(0.5, counter)
+    values.sort(spire.algebra.Order.fromOrdering[Double])
+    var median: Double = Double.NaN
+    if (counter > 0) {
+      median = values(lowerIndex) * (1 - weight) + values(upperIndex) * weight
+      val diffFromMedians: Buffer[Double] = Buffer.ofSize(values.length)
+      val iter = values.iterator()
+      while (iter.hasNext) {
+        diffFromMedians.append(Math.abs(median - iter.next()))
+      }
```

Comment:
Some more weird Scala for-loop syntax (I'm about 90% sure this will work here):
for (value <- values) {
    ...
}